frontend/src/core/services/get-current-user.service.ts

```diff
@@ -1,18 +1,36 @@
 import { Injectable } from '@angular/core';
-import { ReplaySubject, Subject } from 'rxjs';
+import { BehaviorSubject, ReplaySubject, Subject } from 'rxjs';
 import { take } from 'rxjs/operators';
 import { UserService } from 'src/app/user/services/user.service';
+import { LocalStorageKeys } from '../models/local-storage-keys';
 import { UserModel } from '../models/user/user-model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class GetCurrentUserService {
-  private currentUserSubj = new ReplaySubject<UserModel>(1);
-  public currentUser$ = this.currentUserSubj.asObservable();
 
   constructor(private userService: UserService) {}
 
+  private currentUserIdSubj = new BehaviorSubject<number>(this.currentUserId);
+  public currentUserId$ = this.currentUserIdSubj.asObservable();
+
+  public set currentUserId(value: number) {
+      this.currentUserIdSubj.next(value);
+      localStorage.setItem(LocalStorageKeys.selectedUser, value.toString());
+  }
+
+  public get currentUserId(): number {
+      return +(localStorage.getItem(LocalStorageKeys.selectedUser) ?? '-1');
```

Comment:
As i remember we shoud not keep in local storage id of the user. (Not sure). Let it be for know (we have demo, so we must do everything quickly) or better ask our coaches if this approach good.

Reply:
I will fix this after the demo. I asked the coach about it